### DISCUSSION TITLE
module attribute __dict__ restriction in pyarmor 8

### DISCUSSION THREAD

**Discussion by Kadhyo** [dashingsoft/pyarmor#1732] [Q&A] [answered]
hello, i am currently using pyarmor 8.5.2 on python 3.10.4, 
I want to protect my global variables values from being seen by \__dict__ but keep the capacity of my module to be imported, and i can't find a current way to do so,

I found in the documentation that in pyarmor<8 there was a restrict mode '100+' that manage to make the module attribute \__dict__ looks like an empty dictionary, is their a way with pyarmor 8 to have the same effect ?

Thank you

**Comment by jondy** [comment]
Using option `--private` or `--restrict` also could protect the obfusated module attribute `__dict__` in Pyarmor 8. It will raise protection to visit `obfuscated_module.__dict` from plain script.
Please check man page to understand the usage of `--private` and `--restrict`

**Reply by Kadhyo** [reply]
The problem with `--private` or `--restrict` in my use case is that the obfuscated script or package can't be imported by plain script or Python interpreter later

**Comment by jondy** [comment]
It's same as Pyarmor 7, you must obfuscate some modules without `--private`, so that plain scripts could import non-private module.

**Comment by Kadhyo** [comment]
In pyarmor 7 i managed to restrict but keep the ability to import in plain scripts, for example I have a file `foo.py` :
```python
text = 'hello world'
print(text)
```
If I do: 
```
 $ pyarmor-7 obfuscate --restrict=101 foo.py 
```
then 
```
$ cd dist/
$ python 
>>> import foo
hello world
>>> foo.__dict__
{}
```

**Comment by jondy** [comment, marked as answer]
Sorry.
In Pyarmor 8, now the plain scripts could not import private and restrict modules.
In next release, only restrict module could not be imported by plain scripts, private module could be imported by plain scripts, and both of them's `__dict__` are protected.
Thanks for your report
